doc/decisions/frontend_documentation.md

```diff
@@ -0,0 +1,96 @@
+# Frontend Documentation
+
+## Problem
+
+We need to choose a workflow, tooling that helps us to keep the documentation up to date.
+An easy to read and up to date documentation reduces the entry barrier for new-comers.
+Without proper documentation maintenance becomes increasingly more difficult as the project grows.
+
+## Constraints
+
+1. The documentation should be easy to navigate.
+2. The documentation should for the most part, be automatically extracted from the source code.
+
+## Assumptions
+
+1. With a good and beautiful documentation tool, developers are more likely to use and maintain it.
+2. With a documentation tool that enables the viewing of react components, the code quality is improved. Presentation components, should not contain logic with side effects.
+Hooks should be used outside of presentation components, with props passed into them.
+If those rules are not followed, it will be hard to display the components inside the documentation.
+3. With an overview of all presentation components, the whole team can instantly see what is available.
+
+## Considered Alternatives (documentation extraction)
+
+- API-Extractor: [API-Extractor](https://api-extractor.com/) is a tool to extract the public API of a typescript project. It is more suited for libraries.
+
+
+- React Docgen: [React Docgen](https://github.com/reactjs/react-docgen) is a tool from the React Community to extract information from React Components for documentation generation purposes.
+It is not actively maintained anymore.
+It does not generate markdown files, so an additional step is needed, to get just the most basic information.
+
+- JSDoc: [JSDoc](https://jsdoc.app/) is an API documentation generator for JavaScript.
+It is more suited for JavaScript.
+
+## Decision
+We will use [TypeDoc](https://typedoc.org/) with the markdown plugin to extract documentation from the source code.
+
+## Rationale
+TypeDoc has automatic support for most Tags of JSDoc via the TypeScript compiler.
+Additional Tags are supported as well.
+It is easy to configure and invoke and extensible via plugins.
+
+## Considered Alternatives (viewing components)
+
```

Comment:
Docusaurus etc. should be also mentioned. Basically everything that we know and could somewhat do the job should be mentioned.